frontend/src/scenes/annotations/AnnotationModal.tsx

```diff
@@ -110,23 +107,43 @@ export function AnnotationModal({
                                     value: AnnotationScope.Insight,
                                     label: annotationScopeToName[AnnotationScope.Insight],
                                     disabledReason:
-                                        !isInsightScoped && !onSavedInsight && 'You need to save the insight first.',
+                                        (!onSavedInsight && 'You need to save the insight first.') ||
+                                        // if existing annotation data in db (for backwards compatibility) doesn't have insight id set on it
+                                        // we can't let them change scope to insight as we don't know which insight to map to
+                                        (existingModalAnnotation
+                                            ? !existingModalAnnotation?.dashboard_item &&
+                                              "Annotation wasn't originally scoped to an insight so we can't reduce scope to insight level"
+                                            : undefined),
+                                },
+                                {
+                                    value: AnnotationScope.Dashboard,
+                                    label: annotationScopeToName[AnnotationScope.Dashboard],
+                                    tooltip: existingModalAnnotation?.dashboard_name
+                                        ? existingModalAnnotation.dashboard_name
+                                        : undefined,
+                                    disabledReason:
+                                        (!onSavedInsight && 'You need to save the insight first.') ||
+                                        // if existing annotation data in db (for backwards compatibility) doesn't have dashboard id set on it
+                                        // we can't let them change scope to dashboard as we don't know which insight to map to
+                                        (existingModalAnnotation
+                                            ? !existingModalAnnotation?.dashboard &&
+                                              "Annotation wasn't originally scoped to a dashboard so we can't reduce scope to insight level"
+                                            : undefined),
+                                    sideIcon: existingModalAnnotation?.dashboard ? (
+                                        <Link
+                                            to={urls.dashboard(existingModalAnnotation?.dashboard)}
+                                            target="_blank"
+                                            targetBlankIcon
+                                        />
+                                    ) : null,
```

Comment:
This is great – for consistency, we should have an equivalent `sideIcon` for the insight option